File: abdulrhman bzai 2023230604/gym-backend/src/main/java/rs/ac/singidunum/gym/model/MemberModel.java
package rs.ac.singidunum.gym.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;


// It sends only the ids of the plan and trainer, not the whole objects.
@Getter
@Setter
@NoArgsConstructor
public class MemberModel {
    private String name;
    private String email;
    private LocalDate startDate;
    private Integer planId;
    private Integer trainerId; // null = trains alone
}

File: abdulrhman bzai 2023230604/gym-frontend/src/models/member.model.ts
import type { PlanModel } from "./plan.model"
import type { TrainerModel } from "./trainer.model"

export type MemberStatus = 'Active' | 'Expiring soon' | 'Expired'

// What the backend returns: a member together with its plan and trainer (the relations)
export interface MemberModel {
    id: number
    name: string
    email: string
    plan: PlanModel
    trainer: TrainerModel | null   // null = trains alone
    startDate: string
    endDate: string
    totalPrice: number
    status: MemberStatus
    createdAt: string
    updatedAt: string | null
}

// What we send to the backend when creating/updating a member
export interface MemberFormModel {
    name: string
    email: string
    startDate: string
    planId: number | null
    trainerId: number | null
}
